code/mission/missionmessage.cpp

```diff
@@ -316,7 +316,8 @@ int add_avi( char *avi_name )
 	strcpy_s( extra.name, avi_name );
 	strcpy_s( extra.anim_data.filename, avi_name);
 	extra.num = -1;
-	generic_anim_load(&extra.anim_data);
+	generic_anim_load(&extra.anim_data);   // load only to validate the anim
+	generic_anim_unload(&extra.anim_data); // unload to not waste bmpman slots
```

Comment:
The only reason to load the animation data is to validate the animation being used, i.e. that EFFs have all their frames, they have a valid image subtype, etc. The alternative would be to remove both the load/unload (i.e. return it to retail behaviour) however I thought there was a benefit to validating the animations here rather than potential issues being found in-mission.